### DISCUSSION TITLE
[Bug]: Can't do `docker-compose up`, throws error that I need tiktoken

### DISCUSSION THREAD

**Discussion by Auditormadness9** [danny-avila/LibreChat#1025] [Troubleshooting]
### Contact Details

armenemra9@gmail.com

### What happened?

I basically updated LibreChat 1 after I set it up, basically just did git clone into the same repo and overwrote everything, now when I try to compose I get this:

```js
LibreChat         | > LibreChat@0.5.8 backend
LibreChat         | > cross-env NODE_ENV=production node api/server/index.js
LibreChat         |
LibreChat         | Clearing sync timeouts before exiting...
LibreChat         | node:internal/modules/cjs/loader:1093
LibreChat         |   throw err;
LibreChat         |   ^
LibreChat         |
LibreChat         | Error: Cannot find module 'tiktoken/lite'
LibreChat         | Require stack:
LibreChat         | - /app/api/server/utils/countTokens.js
LibreChat         | - /app/api/server/utils/index.js
LibreChat         | - /app/api/server/routes/ask/openAI.js
LibreChat         | - /app/api/server/routes/ask/index.js
LibreChat         | - /app/api/server/routes/index.js
LibreChat         | - /app/api/server/index.js
LibreChat         |     at Module._resolveFilename (node:internal/modules/cjs/loader:1090:15)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:934:27)
LibreChat         |     at Module.require (node:internal/modules/cjs/loader:1157:19)
LibreChat         |     at require (node:internal/modules/helpers:119:18)
LibreChat         |     at Object.<anonymous> (/app/api/server/utils/countTokens.js:1:22)
LibreChat         |     at Module._compile (node:internal/modules/cjs/loader:1275:14)
LibreChat         |     at Module._extensions..js (node:internal/modules/cjs/loader:1329:10)
LibreChat         |     at Module.load (node:internal/modules/cjs/loader:1133:32)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:972:12)
LibreChat         |     at Module.require (node:internal/modules/cjs/loader:1157:19) {
LibreChat         |   code: 'MODULE_NOT_FOUND',
LibreChat         |   requireStack: [
LibreChat         |     '/app/api/server/utils/countTokens.js',
LibreChat         |     '/app/api/server/utils/index.js',
LibreChat         |     '/app/api/server/routes/ask/openAI.js',
LibreChat         |     '/app/api/server/routes/ask/index.js',
LibreChat         |     '/app/api/server/routes/index.js',
LibreChat         |     '/app/api/server/index.js'
LibreChat         |   ]
LibreChat         | }
LibreChat         |
LibreChat         | Node.js v19.9.0
```

I was suggested to run `npm run update:docker`, which I did, but ran into this error:

```js
Starting update script, this may take a minute or two depending on your system and network.
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
Fetching the latest repo...
Switching to main branch...
Already on 'main'
M       package-lock.json
M       package.json
Your branch is behind 'origin/main' by 1 commit, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Pulling the latest code from main...
From https://github.com/danny-avila/LibreChat
 * branch              main       -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
        package.json
Please commit your changes or stash them before you merge.
Aborting
Updating e7ca40b5..d61e4474
node:internal/errors:863
  const err = new Error(message);
              ^

Error: Command failed: git pull origin main
    at checkExecSyncError (node:child_process:885:11)
    at execSync (node:child_process:957:15)
    at C:\LibreChat\config\update.js:77:5
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 12356,
  stdout: null,
  stderr: null
}

Node.js v19.1.0
```

I have also run `npm install tiktoken` but that didn't help with anything.

### Steps to Reproduce

Literally no way I can tell how to reproduce, because what you're basically going to do is do `docker-compose up` and it's gonna work fine for you, but for some reason doesn't for me.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I meant an issues thread on discord but this is fine.

you have changes to the package lock file, you need to discard the changes, then run `npm run update:docker`

**Reply by Auditormadness9** [reply]
I didn't change anything, I cloned the current repo and overwrote everything. What exactly should I discard right now and how exactly?

**Reply by danny-avila** [reply]
At some point you ran `npm install somepqckage` which created a change

**Reply by danny-avila** [reply]
You need git installed and then run `git checkout -- .`

**Reply by Auditormadness9** [reply]
Okay, so what should I discard or how (I don't remember doing anything like that btw)

**Reply by danny-avila** [reply]
Then you can run `npm run update:docker`

**Reply by danny-avila** [reply]
> Okay, so what should I discard or how (I don't remember doing anything like that btw)

you might've followed meep's instructions on discord after you first posted your issue

https://discord.com/channels/1086345563026489514/1086345563026489517/1160259441988472912

**Reply by danny-avila** [reply]
Sorry i realized there was some auto-correct. the correct command to discard unstaged git changes is `git checkout -- .`

**Reply by Auditormadness9** [reply]
![image](https://github.com/danny-avila/LibreChat/assets/71278545/3e9fd51a-2aa9-4d7d-a4c0-b096d41075c4)

**Reply by danny-avila** [reply]
At this point, all I can recommend is doing a fresh installation or using one of the github container images. I just did a fresh installation with no issues (on docker) 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/6d6294d6-160e-4d5d-8de8-ec57fe66e8db)

**Reply by Auditormadness9** [reply]
What about my chats? Will they be saved?

**Reply by danny-avila** [reply]
Keep your old folder and You should copy the /data/ folder from it into the new one

**Reply by Auditormadness9** [reply]
Presumably you mean `meili_data`?

**Reply by Auditormadness9** [reply]
if i bring the old `meili_data` to the new one and overwrite all it throws error during compose. If I bring it over but don't overwrite (skip all conflicts) it stays the same, not recognizing user account.

**Reply by danny-avila** [reply]
LibreChat/data-node/

**Reply by Auditormadness9** [reply]
![image](https://github.com/danny-avila/LibreChat/assets/71278545/663ade7d-9af1-41a8-8ec2-5ac8a47af686)

**Reply by Auditormadness9** [reply]
[log.txt](https://github.com/danny-avila/LibreChat/files/12848919/log.txt)

**Reply by danny-avila** [reply]
what happens when you install? any errors?

**Reply by danny-avila** [reply]
I would be willing to do some live troubleshooting for you if you are comfortable with running some live remote input software like Parsec. DM me on discord for this https://parsec.app/

**Reply by Auditormadness9** [reply]
sure

**Reply by Auditormadness9** [reply]
messaged you on discord
